File: ep7_hw_oop/cpp/oop.cpp
#include <iostream>

using namespace std;

class Car
{
	public:
		Car()
		{
			cout << "Car created." << endl;
		}

		~Car()
		{
			cout << "Car destroyed." << endl;
		}

		void openAshtray()
		{
			cout << "Ashtray opened." << endl;
		}
};

class SpyCar : public Car
{
	public:
		SpyCar()
		{
			cout << "SpyCar created." << endl;
		}

		~SpyCar()
		{
			cout << "SpyCar destroyed." << endl;
		}

		void openAshtray()
		{
			Car::openAshtray();
			cout << "Ejector seat deployed." << endl;
		}
};


int main()
{
	Car car;
	SpyCar spyCar;

	car.openAshtray();

	spyCar.openAshtray();
}

File: ep7_hw_oop/python/oop.py

class Car:
	def __init__( self ):
		print( "Car created." )

	def __del__( self ):
		print( "Car destroyed." )

	def openAshtray( self ):
		print( "Ashtray opened." )

class SpyCar( Car ):
	def __init__( self ):
		print( "SpyCar created." )

	def __del__( self ):
		print( "SpyCar destroyed." )

	def openAshtray( self ):
		super().openAshtray()
		print( "Ejector seat deployed." )

car = Car()
spyCar = SpyCar()

car.openAshtray()
spyCar.openAshtray()
